hoomd/mpcd/SphereGeometry.h

```diff
@@ -55,26 +56,47 @@ class __attribute__((visibility("default"))) SphereGeometry
             {
 
             /*
-             * If particle is still inside the sphere or has zero speed, no collision could have occurred and therefore
-             * exit immediately.
+             * If particle is still inside the sphere , no collision could have occurred and therefore
+             * exit immediately. If particle is on surface, we are assuming it's still inside and
+	     * if it goes outside(during next streaming step) we can backtrack it in the end of next streaming step.
              */
             const Scalar r2 = dot(pos,pos);
             const Scalar v2 = dot(vel,vel);
-            if (r2 <= m_R2 || v2 == Scalar(0))
+	    const Scalar v2_minus_V2 = v2 - m_V2;
+
+            if (r2 <= m_R2)
                {
                dt = Scalar(0);
                return false;
                }
 
             /*
              * Find the time remaining when the particle collided with the sphere of radius R. This time is
-             * found by backtracking the position, r* = r-dt*v, and solving for dt when dot(r*,r*) = R^2.
+             * found by backtracking the position, r* = r-dt*v, and solving for dt when dot(r*,r*) = R'^2.
+	     * where R' is the radius of container when particle collided with spherical geometry (R' = R - V*dt)
              * This gives a quadratic equation in dt; the smaller root is the solution.
              */
 
             const Scalar rv = dot(pos,vel);
-            dt = (rv - fast::sqrt(rv*rv-v2*(r2-m_R2)))/v2;
-
+	    const Scalar RV = m_R*m_V;
+	    const Scalar rv_RV = rv - RV;
+
+	    if(m_V == 0 && v2 == 0)
+	    {
+		    throw std::runtime_error("Velocity of shrinking sphere and velocity of particles is zero");
+	    }
+	    /*dt will be different in the limit v tends to V
+	     *when v2 - V2 ~ 0, different formula(calculated by (lim(v->V)dt)) is used
+	     */
+
+	    if (v2_minus_V2 < Scalar(1e-8) && v2_minus_V2 > -Scalar(1e-8))
+	    {
+		    dt = (r2-m_R2)/(Scalar(2)*rv_RV);
+	    }
+	    else
+	    {
+            dt = (rv_RV - fast::sqrt(rv_RV*rv_RV-v2_minus_V2*(r2-m_R2)))/v2_minus_V2;
```

Comment:
It is important we get this right, so suggest to use the `slow::` implementation in case there is any difference from `fast::`.
```suggestion
            dt = (rv_RV - slow::sqrt(rv_RV*rv_RV-v2_minus_V2*(r2-m_R2)))/v2_minus_V2;
```